mqtt/mqtt-broker/src/sys_topic.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::HashMap;
+
+use bytes::Bytes;
+use mqtt3::proto;
+
+use crate::session::Session;
+use crate::ClientId;
+
+pub enum StateChange<'a> {
```

Comment:
I know this isn't exactly what you originally intended, but I think it makes a lot of sense as a conceptual breakdown of roles. The broker does 2 things. Sends this a state change, and publishes whatever messages come back. The sys_topics file takes changes, and creates the messages needed to notify. 

I don't think returning the state change is good, because some events can trigger multiple messages, which now the broker would have to know about. In addition, my way makes it easy to add more sys topics for things like number of messages sent, b/c it allows 1 event to trigger multiple sys messages.

Reply:
I don't think it is 2 things, it's only one: a broker which publishes state changes. Let's concentrate only on this task for now. And later we can make more agile solution if required.

Reply:
Talked in person, any more comments should be its own thread